File: problems/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        seen: dict[int, int] = {}
        for i, num in enumerate(nums):
            if (j := seen.get(target - num)) is not None:
                return [j, i]
            seen[num] = i
        return []

File: problems/0001-two-sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
  const seen = new Map<number, number>();
  for (const [i, num] of nums.entries()) {
    const j = seen.get(target - num);
    if (j !== undefined) {
      return [j, i];
    }
    seen.set(num, i);
  }
  return [];
}
